library/tasks/MergeSortBam.wdl

```diff
@@ -2,13 +2,43 @@ task MergeSortBamFiles {
   Array[File] bam_inputs
   String sort_order
 
-  Int disk_size = 500
   Int compression_level = 5
 
+  # runtime optional arguments
+  String? opt_docker
```

Comment:
What's the different between 

```wdl
String? opt_docker
String docker = select_first([opt_docker, "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735"])
```
and

```wdl
String docker = "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735"
```

Reply:
functionally the same but the bottom version felt kind of weird to me because if i was looking at the task i wouldnt really think i could provide a value for `docker` because it already has one.  This was definitely the one thing in the whole process that i went back and forth on and just picked a way.  This also plays nicely with in the new `inputs{}` block that will be added in wdl 1.0 (soon) that is supposed to signify things that the user can provide to a task.  Variables not inside the `inputs{}` block can not be overwritten by the caller of the task.  This would look like.

```
task t {
  inputs {
    String? opt_docker
  }
  String docker = select_first([opt_docker, "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735"])
...
}
```
which basically says hey caller, theres a variable `opt_docker` that you can overwrite but `docker` would be hidden from the user.  Similarly the bottom example would translate to

```
task t {
  inputs {
    String docker = "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735"
  }
...
}
```
and I wasnt sure if this made sense visually.  Like your telling the user theres a variable `docker` that you have to set a value for because its not `String?` but it has a default so you dont really have to set a value?  I could go around in circles about which I would rather have.  I chose this way because it has a nice separation between what the user can optionally provide and its defaults but I could be talked into doing it the other way as well.  It all really comes down to what the team decides as there is no "right" way of doing this type of default/overwrite combo.

Reply:
its also very easy to go from one to the other